src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DesignTimeTagHelperWriter.cs

```diff
@@ -32,26 +38,23 @@ public override void WriteAddTagHelperHtmlAttribute(CSharpRenderingContext conte
             context.RenderChildren(node);
         }
 
-        public override void WriteCreateTagHelper(CSharpRenderingContext context, CreateTagHelperIRNode node)
-        {
-            var tagHelperVariableName = GetTagHelperVariableName(node.TagHelperTypeName);
-
-            context.Writer
-                .WriteStartAssignment(tagHelperVariableName)
-                .WriteStartMethodInvocation(
-                    CreateTagHelperMethodName,
-                    "global::" + node.TagHelperTypeName)
-                .WriteEndMethodInvocation();
-        }
-
-        public override void WriteExecuteTagHelpers(CSharpRenderingContext context, ExecuteTagHelpersIRNode node)
-        {
-            // Do nothing
-        }
-
         public override void WriteInitializeTagHelperStructure(CSharpRenderingContext context, InitializeTagHelperStructureIRNode node)
         {
             context.RenderChildren(node);
+
+            // CreateTagHelper
+            foreach (var descriptor in node.TagHelperBinding.Descriptors)
```

Comment:
@NTaylorMullen, I tried moving `CreateTaghelper` logic over to `TagHelperIRNode` rendering but that ends up making things ugly,
```C#
public override void WriteTagHelper(CSharpRenderingContext context, TagHelperIRNode node)
{
    var bodyNode = context.Children.First() as BodyIRNode;
    Debug.Assert(bodyNode != null);
    WriteBodyIRNode(context, bodyNode);

    // CreateTagHelper logic

    // Render attributes
    context.Children.Remove(bodyNode); // Ugly
    context.RenderChildren(node);

    // ExecuteTagHelpers logic
}
```